man/systemd-run-generator.xml

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
+<!--*-nxml-*-->
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN" "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+<!--
+  SPDX-License-Identifier: LGPL-2.1+
+-->
+<refentry id="systemd-run-generator">
+
+  <refentryinfo>
+    <title>systemd-run-generator</title>
+    <productname>systemd</productname>
+  </refentryinfo>
+
+  <refmeta>
+    <refentrytitle>systemd-run-generator</refentrytitle>
+    <manvolnum>8</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>systemd-run-generator</refname>
+    <refpurpose>Generator for invoking commands specified on the kernel command line as system service</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <para><filename>/usr/lib/systemd/system-generators/systemd-run-generator</filename></para>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><filename>systemd-run-generator</filename> is a generator
+    that reads the kernel command line and understands three
+    options:</para>
+
+    <para>If the <option>systemd.run=</option> option is specified and followed by a command line, a unit named
+    <filename>kernel-command-line.service</filename> is generated for it and booted into. The service has
+    <varname>Type=oneshot</varname> set, and has <varname>SuccessAction=exit</varname> and
+    <varname>FailureAction=exit</varname> configured by default, thus ensuring that the system is shut down as soon as
+    the command completes. The exit status of the command line is propagated to the invoking container manager, if
+    this applies (which might propagate this further, to the calling shell — e.g.
+    <citerefentry><refentrytitle>systemd-nspawn</refentrytitle><manvolnum>7</manvolnum></citerefentry> does this). If
+    this option is used multiple times the unit file will contain multiple <varname>ExecStart=</varname> lines, to
+    execute all commands in order. The command is started as regular service, i.e. with
+    <varname>DefaultDependencies=</varname> on. </para>
```

Comment:
Hmm, if the run command is invalid, the unit will not be started, and the machine is left "hanging".
```sudo ~/src/systemd-work3/build/systemd-nspawn -M rawhide -n -b --bind=/home 'systemd.run=foobar'
...
/run/systemd/generator/kernel-command-line.service:10: Executable "foobar" not found in path "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
[  OK  ] Reached target Command from Kernel Command Line.
(hangs doing nothing)
```
Not sure what to do about this.

Reply:
I changed the generator now to emit Requires= rather than Wants from the generated target to the generated service. THis means the initial transaction already fails at the moment it is put together, which causes the service manager to complain very loudly and then going to freeze mode.

I am pretty sure it shouldn't go into freeze mode when running in a container (it does sound appropriate when run on a physical machine though), but that's something to fix in a separate PR I figure.